Lesson2/SecondHomework.playground/Contents.swift

```diff
@@ -0,0 +1,37 @@
+/// An object, which is a list of elements with a type integer, organized according to the LIFO principle.
+class Stack {
+    private var array: [Int] = [] {
+        didSet {
+            minimalValue = array.min()
+        }
+    }
+    private var minimalValue: Int?
+    
+    func push(element: Int) {
+        array.append(element)
+    }
+    
+    func pop() -> Int? {
+        if array.isEmpty {
+            return nil
+        } else {
+            let lastElement = array.removeLast()
+            return lastElement
+        }
+    }
+    
+    func top() -> Int? {
+        return array.last
+    }
+    
+    func min() -> Int? {
```

Comment:
The `Stack` should not have `min()` function in the interface.

In general, the base class `Stack` should be as simple as possible.

Reply:
According to your current implementation, in case I want to update `StackStatistics` with a method `maximumElement()`, I will have to update `Stack`'s implementation too:
- add `max()` method to `Stack`;
- update `didSet` observers in `Stack`;
- implement `maximumElement()` for `StackStatistics`;

I should be able to extend `StackStatistics` without modifying the original base class.

Reply:
Made changes to the Stack and StackStatistics classes